iceoryx2-bb/lock-free/src/mpmc/container.rs

```diff
@@ -452,6 +492,34 @@ impl<T: Copy + Debug, const CAPACITY: usize> FixedSizeContainer<T, CAPACITY> {
         unsafe { self.state.add(value) }
     }
 
+    /// Adds a new element to the [`FixedSizeContainer`]. If there is no more space available it returns
+    /// [`None`], otherwise [`Some`] containing the the index value to the underlying element.
+    ///
+    /// ```
+    /// use iceoryx2_bb_lock_free::mpmc::container::*;
+    ///
+    /// const CAPACITY: usize = 139;
+    /// let container = FixedSizeContainer::<u128, CAPACITY>::new();
+    ///
+    /// match unsafe { container.add_raw(1234567) } {
+    ///     Some(index) => {
+    ///         println!("added at index {}", index);
+    ///         unsafe { container.remove_raw_index(index) };
+    ///     },
+    ///     None => println!("container is full"),
+    /// };
+    ///
+    /// ```
+    ///
+    /// # Safety
+    ///
+    ///  * Use [`FixedSizeContainer::remove_raw_index()`] to release the acquired index again. Otherwise,
+    ///     the element will leak.
```

Comment:
Food for thought. I guess `remove_raw_index` does not only remove the index but the corresponding value. Saying that, wouldn't `remove_with_raw_index` be more clear?